lib/internal/modules/esm/resolve.js

```diff
@@ -595,13 +493,15 @@ function packageResolve(specifier, base, conditions) {
     }
 
     // Package match.
-    const packageConfig = getPackageConfig(packageJSONPath, base);
+    const packagePath = StringPrototypeReplace(
+      packageJSONPath, `${sep}package.json`, '');
```

Comment:
I believe this will actually replace eg `/path/to/package.json/package.json` with `/path/to//package.json`.

Rather just trim the end off since it's guaranteed to end in `/package.json` - `packagePath.slice(0, -'/package.json'.length)` sort of thing.